src/DotNetty.Transport/Channels/AbstractChannel.cs

```diff
@@ -188,27 +195,6 @@ public virtual Task CloseAsync()
 
         public Task DeregisterAsync()
         {
-            /// <summary>
-            /// One problem of channel deregistration is that after a channel has been deregistered
-            /// there may still be tasks, created from within one of the channel's ChannelHandlers,
-            /// input the {@link EventLoop}'s task queue. That way, an unfortunate twist of events could lead
-            /// to tasks still being input the old {@link EventLoop}'s queue even after the channel has been
-            /// registered with a new {@link EventLoop}. This would lead to the tasks being executed by two
-            /// different {@link EventLoop}s.
-            ///
-            /// Our solution to this problem is to always perform the actual deregistration of
-            /// the channel as a task and to reject any submission of new tasks, from within
-            /// one of the channel's ChannelHandlers, until the channel is registered with
-            /// another {@link EventLoop}. That way we can be sure that there are no more tasks regarding
-            /// that particular channel after it has been deregistered (because the deregistration
-            /// task is the last one.).
-            ///
-            /// This only works for one time tasks. To see how we handle periodic/delayed tasks have a look
-            /// at {@link io.netty.util.concurrent.ScheduledFutureTask#run()}.
-            ///
-            /// Also see {@link HeadContext#deregister(ChannelHandlerContext, ChannelPromise)}.
-            /// </summary>
-            this.eventLoop.RejectNewTasks();
```

Comment:
DotNetty was originally based on netty 5.0. That got dropped since and 4.1 is where things are now. This is part of changes that got dropped for a reason of "no clear improvement".